src/components/v2/Toggle/index.tsx

```diff
@@ -1,19 +1,23 @@
+/** @jsxImportSource @emotion/react */
 import React from 'react';
 import Switch from '@mui/material/Switch';
+import { SerializedStyles } from '@emotion/react';
+import { SwitchBaseProps } from '@mui/material/internal/SwitchBase';
 import { useStyles } from './styles';
 
 export interface IToggleProps {
-  onChange: (event: React.ChangeEvent<HTMLInputElement>, checked: boolean) => void;
+  onChange: SwitchBaseProps['onChange'];
   value: boolean;
+  parentStyles?: SerializedStyles;
 }
 
 const label = { inputProps: { 'aria-label': 'Switch' } };
 
-export const Toggle = ({ onChange, value }: IToggleProps) => {
+export const Toggle = ({ onChange, value, parentStyles }: IToggleProps) => {
   const styles = useStyles();
   return (
     <Switch
-      sx={styles}
+      css={[styles, parentStyles]}
```

Comment:
I think you're getting confused with the styling.
You don't need to pass any css prop in order to extend the styles of the component. Instead, you can just pass the `className` prop to the component. This will have the exact same result:

```typescript
export interface IToggleProps {
  onChange: SwitchBaseProps['onChange'];
  value: boolean;
  className?: string;
}

export const Toggle = ({ onChange, value, className }: IToggleProps) => {
  ...

  return (
    <Switch 
      css={styles}
      className={className}
      ...
```

Doing that, your component can still get its styles extended, except it can be done through the already supported `css` prop:

```typescript
<Toggle css={yourCustomStyle} />
```

I understand it's confusing, it's an Emotion thing :/